File: UI/src/lib/types/team.ts
export interface Team {
	id: string;
	name: string;
	key: string;
	description: string | null;
	color: string | null;
	icon: string | null;
	triage_enabled: boolean;
	parent_auto_close_enabled: boolean;
	sub_issue_auto_close_enabled: boolean;
	issue_copy_prompt: string | null;
	created_at: string;
	updated_at: string;
}

File: BE/internal/dto/team.go
package dto

import "time"

type CreateTeamRequest struct {
	Name        string  `json:"name" validate:"required,min=1,max=100"`
	Key         string  `json:"key" validate:"required,min=1,max=10,alpha,uppercase"`
	Description *string `json:"description"`
	Color       *string `json:"color"`
	Icon        *string `json:"icon"`
}

type UpdateTeamRequest struct {
	Name          *string `json:"name" validate:"omitempty,min=1,max=100"`
	Description   *string `json:"description"`
	Color         *string `json:"color"`
	Icon          *string `json:"icon"`
	EstimateScale *string `json:"estimate_scale" validate:"omitempty,oneof=linear exponential fibonacci tshirt"`
}

type TeamResponse struct {
	ID            string    `json:"id"`
	Name          string    `json:"name"`
	Key           string    `json:"key"`
	Description   *string   `json:"description"`
	Color         *string   `json:"color"`
	Icon          *string   `json:"icon"`
	EstimateScale string    `json:"estimate_scale"`
	CreatedAt     time.Time `json:"created_at"`
	UpdatedAt     time.Time `json:"updated_at"`
}
